src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -807,63 +807,206 @@ ErrorIfInsertPartitionColumnDoesNotMatchSelect(Query *query, RangeTblEntry *inse
 	uint32 rangeTableId = 1;
 	Oid insertRelationId = insertRte->relid;
 	Var *insertPartitionColumn = PartitionColumn(insertRelationId, rangeTableId);
-	bool partitionColumnsMatch = false;
 	Query *subquery = subqueryRte->subquery;
+	bool targetTableHasPartitionColumn = false;
 
 	foreach(targetEntryCell, query->targetList)
 	{
 		TargetEntry *targetEntry = (TargetEntry *) lfirst(targetEntryCell);
+		List *insertTargetEntryColumnList = pull_var_clause_default((Node *) targetEntry);
+		int insertTargetEntryColumnCount = list_length(insertTargetEntryColumnList);
+		Var *insertVar = NULL;
+		AttrNumber originalAttrNo = InvalidAttrNumber;
+		TargetEntry *subqeryTargetEntry = NULL;
 
-		if (IsA(targetEntry->expr, Var))
+		/*
+		 * We only consider target entries that include a single column. Note that this
```

Comment:
Oh, it looks like I forgot to mention the case where `insertTargetEntryColumnCount  = 0`.

If you include DEFAULT values in the table such as:
```SQL
create table t2 (a int DEFAULT 5, b int, c int);
 select create_distributed_table('t2', 'c');
insert into t2 (b,c) select b,c FROM t2;
```

The `insertTargetEntryColumnCount` becomes `0` for column `a`. I'll update the comment and add a regression test.